backend/src/apiserver/storage/list_util.go

```diff
@@ -15,77 +15,43 @@
 package storage
 
 import (
-	"encoding/base64"
-	"encoding/json"
+	"database/sql"
 	"fmt"
-	"reflect"
 
 	sq "github.com/Masterminds/squirrel"
 	"github.com/kubeflow/pipelines/backend/src/apiserver/common"
-	"github.com/kubeflow/pipelines/backend/src/apiserver/model"
 	"github.com/kubeflow/pipelines/backend/src/common/util"
 )
 
-// Delegate to query the model table for a list of models.
-type QueryListableModelTable func(request *common.PaginationContext) ([]model.ListableDataModel, error)
-
-// Logic for listing ListableModels. This would attempt to query one more model than requested,
-// in order to generate the page token. If the result is less than request+1, the nextPageToken
-// will be empty
-func listModel(context *common.PaginationContext, queryTable QueryListableModelTable) (models []model.ListableDataModel, nextPageToken string, err error) {
-	newContext := *context
-	// List one more item to generate next page token.
-	newContext.PageSize = context.PageSize + 1
-	results, err := queryTable(&newContext)
-	if err != nil {
-		return nil, "", util.Wrap(err, "List data model failed.")
-	}
-	if len(results) < newContext.PageSize {
-		return results, "", nil
+func FilterOnResourceReference(tableName string, resourceType common.ResourceType, selectCount bool,
```

Comment:
how about passing "*" and "count(*)" string instead of selectCount boolean flag. maybe consider define them as const? it would be more readable from method caller perspective.

Reply:
I can do that, but `selectCount` is passed through from the different implementations of `buildSelectXXXQuery`. I think that's a better interface there, since it's being relied on to change the way the query is built. If you really want this change, do you think it should be propagated to all 4 implementations?